stringfuzz/transformers/rotate.py

```diff
@@ -0,0 +1,77 @@
+import random
+import string
+
+from stringfuzz.ast import *
+from stringfuzz.ast_walker import ASTWalker
+
+__all__ = [
+    'rotate',
+]
+
+class RotateTransformer(ASTWalker):
+    def __init__(self, ast):
+        super(RotateTransformer, self).__init__(ast)
+
+    @property
+    def ast(self):
+        return self._ASTWalker__ast
+
+    def exit_expression(self, expr):
+        if isinstance(expr, ConcatNode):
+            if isinstance(expr.body[0], ConcatNode):
+                # rotate clockwise ll, lr, r
+                temp = expr.body[1]
+                expr.body[1] = expr.body[0].body[0]
+                expr.body[0].body[0] = expr.body[0].body[1]
+                expr.body[0].body[1] = temp
+            if isinstance(expr.body[1], ConcatNode):
+                # rotate clockwise l, rl, rr
+                temp = expr.body[0]
+                expr.body[0] = expr.body[1].body[0]
+                expr.body[1].body[0] = expr.body[1].body[1]
+                expr.body[1].body[1] = temp   
+        elif isinstance(expr, ReConcatNode):
+            if isinstance(expr.body[0], ReConcatNode):
+                # rotate clockwise ll, lr, r
+                temp = expr.body[1]
+                expr.body[1] = expr.body[0].body[0]
+                expr.body[0].body[0] = expr.body[0].body[1]
+                expr.body[0].body[1] = temp
+            if isinstance(expr.body[1], ReConcatNode):
+                # rotate clockwise l, rl, rr
+                temp = expr.body[0]
+                expr.body[0] = expr.body[1].body[0]
+                expr.body[1].body[0] = expr.body[1].body[1]
+                expr.body[1].body[1] = temp
+        elif isinstance(expr, ReUnionNode):
+            if isinstance(expr.body[0], ReUnionNode):
+                # rotate clockwise ll, lr, r
+                temp = expr.body[1]
+                expr.body[1] = expr.body[0].body[0]
+                expr.body[0].body[0] = expr.body[0].body[1]
+                expr.body[0].body[1] = temp
+            if isinstance(expr.body[1], ReUnionNode):
+                # rotate clockwise l, rl, rr
+                temp = expr.body[0]
+                expr.body[0] = expr.body[1].body[0]
+                expr.body[1].body[0] = expr.body[1].body[1]
+                expr.body[1].body[1] = temp
+        elif isinstance(expr, InReNode):
+            if isinstance(expr.body[0], InReNode):
+                # rotate clockwise ll, lr, r
+                temp = expr.body[1]
+                expr.body[1] = expr.body[0].body[0]
+                expr.body[0].body[0] = expr.body[0].body[1]
+                expr.body[0].body[1] = temp
+            if isinstance(expr.body[1], InReNode):
+                # rotate clockwise l, rl, rr
+                temp = expr.body[0]
+                expr.body[0] = expr.body[1].body[0]
+                expr.body[1].body[0] = expr.body[1].body[1]
+                expr.body[1].body[1] = temp
```

Comment:
Do you think it might be helpful to factor the first level of `if/elif`s into specific enter/exit methods in `ASTWalker` for later reuse? Or would it be the same amount of code/complexity either way?

Reply:
It would probably be cleaner! Especially if the enter/exit methods are based on a notion of types. I'll think about this a bit more and come back with a more concrete proposal.

Reply:
Ok, sounds good!